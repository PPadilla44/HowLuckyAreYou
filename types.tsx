import { NavigatorScreenParams } from '@react-navigation/native';
import { NativeStackScreenProps } from '@react-navigation/native-stack';

export interface OddsItemInterface {
    id: string
    title: string;
    displayOdds: string,
    oddsString?: string;
    fraction?: {
        denominator: string;
        numerator: string;
    };
    multiplier: string;
}

declare global {
    namespace ReactNavigation {
        interface RootParamList extends RootStackParamList { }
    }
}

export type RootStackParamList = {
    Root: NavigatorScreenParams<RootTabParamList> | undefined;
    Modal: undefined;
    NotFound: undefined;
};

export type RootStackScreenProps<Screen extends keyof RootStackParamList> = NativeStackScreenProps<
    RootStackParamList,
    Screen
>;


export type RootTabParamList = {
    Main: undefined;
    Modal: undefined;
};

export type RootTabScreenProps<Screen extends keyof RootTabParamList> = NativeStackScreenProps<RootStackParamList>;
