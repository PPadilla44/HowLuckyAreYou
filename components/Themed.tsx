import { Text as DefaultText, View as DefaultView, ScrollViewProps as dScrollViewProps } from 'react-native';
import { Icon as DefaultIcon, IconProps as DefaultIconProps, Input as DefaultInput, InputProps as DefaultInputProps, ButtonGroup as DButtonGroup, ButtonGroupProps as DButtonGroupProps } from 'react-native-elements';
import { ScrollView as DefaultScrollView, NativeViewGestureHandlerProps, TouchableOpacity as DefaultTouchableOpacity } from "react-native-gesture-handler";

import Colors from '../constants/Colors';

import useColorScheme from '../hooks/useColorScheme';

export const useThemeColor = (
    props: { light?: string; dark?: string },
    colorName: keyof typeof Colors.light & keyof typeof Colors.dark
) => {
    // const theme = useColorScheme();
    const theme = "dark";

    const colorFromProps = props[theme];

    return { colorFromProps: colorFromProps ? colorFromProps : Colors[theme][colorName], theme }
}

type ThemeProps = {
    lightColor?: string;
    darkColor?: string;
};

export type TextProps = ThemeProps & DefaultText['props'];
export type ViewProps = ThemeProps & DefaultView['props'];
export type TouchableOpacityProps = ThemeProps & DefaultTouchableOpacity['props'];
export type IconProps = ThemeProps & DefaultIconProps;
export type InputProps = ThemeProps & DefaultInputProps;
export type ScrollViewProps = ThemeProps & NativeViewGestureHandlerProps & dScrollViewProps;
export type ButtonGroupProps = ThemeProps & DButtonGroupProps;

export const Text = (props: TextProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "text").colorFromProps;

    return <DefaultText style={[{ color, fontFamily: "Futura" }, style]} {...otherProps} />
}

export const View = (props: ViewProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "background").colorFromProps;
    const shadowColor = useThemeColor({  }, "text").colorFromProps;
    return <DefaultView style={[{ backgroundColor, shadowColor }, style]}  {...otherProps} />
}

export const TouchableOpacity = (props: TouchableOpacityProps) => {
    const { style, containerStyle, lightColor, darkColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "button").colorFromProps;
    const oppositeColor = useThemeColor({ }, "text").colorFromProps;
    return <DefaultTouchableOpacity style={[{ backgroundColor, borderColor: oppositeColor, shadowColor: oppositeColor }, style]} containerStyle={[{ backgroundColor, borderColor: oppositeColor }, containerStyle]} {...otherProps} />
}

export const Icon = (props: IconProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "icon").colorFromProps;
    return <DefaultIcon tvParallaxProperties={undefined} color={color} {...otherProps} />
}

export const Input = (props: InputProps) => {
    const { style, lightColor, darkColor, ...otherProps } = props;
    const { colorFromProps, theme } = useThemeColor({ light: lightColor, dark: darkColor }, "input");
    const backgroundColor = colorFromProps;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "text").colorFromProps;
    return (

        <>
            <DefaultInput
                style={[{ backgroundColor: "transparent", color }, style]}
                keyboardAppearance={theme}
                {...otherProps}
            />
            <View style={{ backgroundColor, width: "100%", height: "100%", opacity: .5, position: "absolute", borderRadius: 15, zIndex: -10 }} />
        </>
    )
    // return <DefaultInput style={[{backgroundColor: "transparent", color }, style]} keyboardAppearance={theme} {...otherProps} />
}

export const ScrollView = (props: ScrollViewProps) => {
    const { style, darkColor, lightColor, ...otherProps } = props;
    const backgroundColor = useThemeColor({ light: lightColor, dark: darkColor }, "background").colorFromProps;
    return <DefaultScrollView style={[{ backgroundColor }, style]} {...otherProps} />
}

export const ButtonGroup = (props: ButtonGroupProps) => {
    const { containerStyle, darkColor, lightColor, ...otherProps } = props;
    const color = useThemeColor({ light: lightColor, dark: darkColor }, "input").colorFromProps;
    
    return <DButtonGroup containerStyle={[{ backgroundColor: color, borderColor: color }, containerStyle]}  {...otherProps} />
}