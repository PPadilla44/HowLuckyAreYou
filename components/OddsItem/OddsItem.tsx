import { StyleSheet } from 'react-native'
import React, { FC, useRef } from 'react'
import { Swipeable } from 'react-native-gesture-handler'
import { OddsItemInterface } from '../../types'
import RightSwipe from './RightSwipe'
import LeftSwipe from './LeftSwipe/LeftSwipe';
import { Text, TouchableOpacity } from "../Themed"
import Colors from '../../constants/Colors'
import { removeItem } from '../../store/utils/thunkerFunctions'
import { useOddsItems } from '../contexts/useOddsItems'
import { useClicker } from '../contexts/useClicker'
import { useNavigation } from '@react-navigation/native'

interface Props {
    item: OddsItemInterface
}


const OddsItem: FC<Props> = ({ item }) => {

    const { state, dispatch } = useOddsItems();
    const { dispatch: cDispatch } = useClicker();
    const navigation = useNavigation();

    const ref = useRef<Swipeable>(null)

    const deleteItem = async (id: string) => {
        removeItem(state, dispatch, id);
        ref.current?.state.dragX.setValue(0)
        ref.current?.state.rowTranslation.setValue(0);
    }

    const setDisplay = (oddItem: OddsItemInterface) => {
        cDispatch!({ type: 'SET_DISPLAY', payload: oddItem })
        navigation.goBack();
    }

    return (
        <Swipeable
            ref={ref}
            // renderLeftActions={() => <LeftSwipe callback={() => console.log("LEFTSWIPE")} />}
            renderRightActions={() => <RightSwipe callback={() => deleteItem(item.id)} />}
        >
            <TouchableOpacity testID='oddsItemBtn' lightColor='white' darkColor={Colors.dark.modal} onPress={() => setDisplay(item)} style={styles.item} containerStyle={styles.itemContainer} activeOpacity={.8}>
                <Text style={styles.baseTxt}>{item.title}</Text>
                <Text style={styles.baseTxt}>
                    {
                        item.multiplier === "B" || item.multiplier === "M" ?
                            `${item.fraction?.numerator} / ${item.fraction?.denominator} ${item.multiplier}`
                            :
                            parseInt(item.fractionPref) ?
                                `${item.fraction?.numerator} / ${parseInt(item.fraction!.denominator) * parseInt(item.multiplier)}`
                                :
                                `${item.oddsString}%`
                    }
                </Text>
            </TouchableOpacity>
        </Swipeable>
    )
}

export default OddsItem

const styles = StyleSheet.create({
    itemContainer: {
        paddingHorizontal: 30,
        paddingVertical: 20,
    },
    item: {
        flexDirection: "row",
        justifyContent: "space-between",
    },
    baseTxt: {
        fontSize: 24
    }
})