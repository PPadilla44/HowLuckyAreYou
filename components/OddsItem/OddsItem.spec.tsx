import { cleanup, fireEvent, render, RenderAPI } from "@testing-library/react-native";
import React from "react";
import { OddsItemInterface } from "../../types";
import { Text } from "../Themed";
import OddsItem from "./OddsItem";

describe('<OddsItem />', () => {
    
    const props : OddsItemInterface = {
        id: "1",
        name: "test1",
        odds: "90%"
    } 

    let wrapper: RenderAPI;

    afterEach(cleanup)

    beforeEach(() => {
        wrapper = render(<OddsItem item={props} /> )
    });

    it('should render correctly', () => {
        wrapper.getByText("test1");
        wrapper.getByText("90%");
        expect(wrapper.container.findAllByType(Text)).toHaveLength(2)
        wrapper.getByTestId("oddsItemBtn");
    });

    it('should render press on item', () => {
        const logSpy = jest.spyOn(console, "log");
        fireEvent.press(wrapper.getByTestId("oddsItemBtn"));
        expect(logSpy).toBeCalled()
    });

    it('should render correct styling', () => {
        expect(wrapper.getByTestId("oddsItemBtn")).toHaveStyle({
            paddingHorizontal: 30,
            paddingVertical: 20,
            flexDirection: "row",
            justifyContent: "space-between",
        });

        expect(wrapper.container.findAllByType(Text)[0]).toHaveStyle({
            fontSize: 24
        })
        expect(wrapper.container.findAllByType(Text)[1]).toHaveStyle({
            fontSize: 24
        })
    });

    it('should match snapshot', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });


});