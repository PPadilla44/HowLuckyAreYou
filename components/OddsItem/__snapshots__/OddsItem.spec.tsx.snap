// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OddsItem /> should match snapshot 1`] = `
<View
  collapsable={false}
  forwardedRef={[Function]}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  onLayout={[Function]}
  style={
    Object {
      "overflow": "hidden",
    }
  }
>
  <View
    style={
      Object {
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateX": -10000,
          },
        ],
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      exclusive={true}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Object {
          "alignItems": "flex-start",
          "backgroundColor": "#325C99",
          "justifyContent": "center",
        }
      }
      testID="leftSwipeBtn"
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "600",
              "paddingHorizontal": 30,
              "paddingVertical": 20,
            }
          }
        >
          Share
        </Text>
      </View>
    </RNGestureHandlerButton>
    <View
      onLayout={[Function]}
    />
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "flexDirection": "row-reverse",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "transform": Array [
          Object {
            "translateX": -10000,
          },
        ],
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      exclusive={true}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "red",
          "justifyContent": "center",
        }
      }
      testID="rightSwipeBtn"
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontWeight": "bold",
              "paddingHorizontal": 30,
              "paddingVertical": 20,
            }
          }
        >
          Delete
        </Text>
      </View>
    </RNGestureHandlerButton>
    <View
      onLayout={[Function]}
    />
  </View>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    pointerEvents="auto"
    style={
      Object {
        "transform": Array [
          Object {
            "translateX": -0,
          },
        ],
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      exclusive={true}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "backgroundColor": "#252525",
            "borderColor": "white",
          },
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingHorizontal": 30,
            "paddingVertical": 20,
          },
        ]
      }
      testID="oddsItemBtn"
    >
      <View
        accessible={true}
        style={
          Object {
            "backgroundColor": "#252525",
            "borderColor": "white",
            "opacity": 1,
            "shadowColor": "white",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "Futura",
              },
              Object {
                "fontSize": 24,
              },
            ]
          }
        >
          test1
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "Futura",
              },
              Object {
                "fontSize": 24,
              },
            ]
          }
        >
          90%
           
        </Text>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;
